libs/gls_cli.py

```diff
@@ -0,0 +1,447 @@
+    )
+    parser.add_argument(
+        "-g",
+        "--gene-corr-file",
+        required=False,
+        help="Path to a gene correlations file or folder. It's is mandatory if "
+        "running a GLS model, and not necessary for OLS.",
+    )
+    parser.add_argument(
+        "--debug-use-ols",
+        required=False,
+        action="store_true",
+        help="It uses a standard OLS model instead of GLS. For debugging " "purposes.",
+    )
+    parser.add_argument(
+        "--debug-use-sub-gene-corr",
+        required=False,
+        action="store_true",
+        help="It uses an LV-specific submatrix of the gene correlation matrix.",
+    )
+    parser.add_argument(
+        "-l",
+        "--lv-list",
+        required=False,
+        nargs="+",
+        default=[],
+        help="List of LV (gene modules) identifiers on which an association "
+        "will be computed. All the rest not in the list are ignored.",
+    )
+    parser.add_argument(
+        "--covars",
+        required=False,
+        nargs="+",
+        choices=COVAR_OPTIONS,
+        help="List of covariates to use.",
+    )
+    parser.add_argument(
+        "--cohort-metadata-dir",
+        required=False,
+        type=str,
+        help="Directory where cohort metadata files are stored.",
+    )
+    parser.add_argument(
+        "--batch-id",
+        required=False,
+        type=int,
+        help="With --batch-n-splits, it allows to distribute computation of "
+        "each LV-trait pair across a set of batches",
+    )
+    parser.add_argument(
+        "--batch-n-splits",
+        required=False,
+        type=int,
+        help="With --batch-id, it allows to distribute computation of "
+        "each LV-trait pair across a set of batches",
+    )
+    parser.add_argument(
+        "--duplicated-genes-action",
+        required=False,
+        choices=["keep-first", "keep-last", "remove-all"],
+        help="Mandatory if gene identifies are duplicated.",
+    )
+
+    args = parser.parse_args()
+
+    # check compatibility of parameteres
+    if len(args.lv_list) > 0 and (
+        args.batch_id is not None or args.batch_n_splits is not None
+    ):
+        logger.error(
+            "Incompatible parameters: LV list and batches cannot be used together"
+        )
+        sys.exit(2)
+
+    if (args.batch_id is not None and args.batch_n_splits is None) or (
+        args.batch_id is None and args.batch_n_splits is not None
+    ):
+        logger.error(
+            "Both --batch-id and --batch-n-splits have to be provided (not only one of them)"
+        )
+        sys.exit(2)
+
+    if args.batch_id is not None and args.batch_id < 1:
+        logger.error("--batch-id must be >= 1")
+        sys.exit(2)
+
+    if (
+        args.batch_id is not None
+        and args.batch_n_splits is not None
+        and args.batch_id > args.batch_n_splits
+    ):
+        logger.error("--batch-id must be <= --batch-n-splits")
+        sys.exit(2)
```

Comment:
Is there a reason you sometimes use exit code 2 and sometimes use exit code 1? What are their different meanings?

Reply:
Good point. Honestly, I don't remember if there is a good reason. I think I tried to separate different outcomes. It would be risky to change that now because many scripts call this code. So I created an issue [here](https://github.com/greenelab/phenoplier/issues/66).

Thank you, Ben!